src/components/HomeSearchBox/index.tsx

```diff
@@ -1,42 +1,110 @@
-import React, { useState } from 'react'
+import React, { useState, useEffect } from 'react'
 import { useRouter } from 'next/router'
 import useTranslation from 'next-translate/useTranslation'
 
 const HomeSearchBox = () => {
+  const [searchValue, setSearchValue] = useState<string>('')
+  const [active, setActive] = useState(0)
+  const [filteredSuggestion, setFilteredSuggestion] = useState<string[]>([])
+  const [isSuggestionShown, setSuggestionShown] = useState(false)
   const { t } = useTranslation('home')
   const router = useRouter()
-  const [searchValue, setSearchValue] = useState<string>('')
+
+  useEffect(() => {
+    const handleMouseClick = (event: MouseEvent) => {
+      setSuggestionShown(false)
+    }
+
+    document.addEventListener('mousedown', handleMouseClick)
+
+    return () => {
+      document.removeEventListener('mousedown', handleMouseClick)
+    }
+  }, [])
 
   const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setSearchValue(event.target.value)
+    //TODO: This below data should be obtained from database
+    const suggestions: string[] = ['Alabama', 'Alaska', 'American Samoa', 'Bubble Holmes', 'Contra Hits', 'Amazon']
+
+    const {
+      currentTarget: { value }
+    } = event
+    const newFilteredSuggestionSuggestions = suggestions.filter(
+      (suggestion: string) => suggestion.toLowerCase().indexOf(value.toLowerCase()) > -1
+    )
+
+    setActive(0)
+    setFilteredSuggestion(newFilteredSuggestionSuggestions)
+    setSuggestionShown(true)
+    setSearchValue(value)
+  }
+
+  const handleListItemClick = (event: React.MouseEvent<HTMLLIElement>) => {
+    setActive(0)
+    setFilteredSuggestion([])
+    setSuggestionShown(false)
+    setSearchValue(event.currentTarget.innerText)
+    handleSearchClick()
   }
 
   const handleSearchClick = () => {
     if (searchValue) {
-      router.push('products/${searchValue}')
+      router.push(`products/${searchValue}`)
       setSearchValue('')
     }
   }
 
   const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
-    if (event.key === 'Enter') {
+    if (searchValue && event.key === 'Enter') {
+      setActive(0)
+      setSuggestionShown(false)
+      setSearchValue(filteredSuggestion[active])
       handleSearchClick()
     }
   }
 
+  const renderSearchSuggestion = () => {
+    if (isSuggestionShown && searchValue) {
+      const filterResults = filteredSuggestion.map((suggestion, index) => {
+        let className = index === active ? 'active' : ''
+
+        return (
+          <li className={className} key={suggestion} onClick={handleListItemClick}>
+            {suggestion}
+          </li>
+        )
+      })
+
+      const hasFilteredSuggestionSuggestion: boolean = filteredSuggestion.length > 0
+      const autoComplete = hasFilteredSuggestionSuggestion ? (
+        <ul className="home-search-suggestion__autocomplete">{filterResults}</ul>
```

Comment:
I have 2 concerns here
1. We have search box in our Product list page also, should we remove this `home` prefix?
2. Normally, we name our css class by its `HTML` tag. Do you think `list` is better than `autocomplete`?

```suggestion
        <ul className="search-suggestion__list">{filterResults}</ul>
```